### ISSUE TITLE
Recent commit broke instance_tags in ec2 module

### ISSUE THREAD

**Issue by dekked** [ansible/ansible#4753]
This is the commit: https://github.com/ansible/ansible/commit/90cce35919e6b93c0b904078c5285e1dbaeb0fdf

The following `site.yml` file used to work with the previous commit. It launches an ec2 instance. You are probably going to need to modify `keypair`, `region` and `ami_id` variables to reproduce, and also need to have the `AWS_ACCESS_KEY` and `AWS_SECRET_KEY` environment variables present.

```

---

- name: test
  hosts: all
  connection: local

  vars:

    keypair: key-test
    ami_id: ami-3a784d7f
    region: us-west-1
    instance_tags:
      Name: test instance
      other: tag

  tasks:

    - name: launch instance
      local_action:
        module: ec2
        keypair: "{{ keypair }}"
        instance_type: t1.micro
        region: "{{ region }}"
        image: "{{ ami_id }}"
        instance_tags: "{{ instance_tags|to_json }}"
        wait: yes
        wait_timeout: 500
        count: 1
```

Currently, this happens when I run it:

``` python
 _____________
< PLAY [test] >
 -------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


 _________________
< GATHERING FACTS >
 -----------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


ok: [127.0.0.1]
 _______________________
< TASK: launch instance >
 -----------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


failed: [127.0.0.1] => {"failed": true, "parsed": false}
invalid output was: Traceback (most recent call last):
  File "/home/alan/.ansible/tmp/ansible-1383244452.89-240799362355153/ec2", line 1587, in <module>
    main()
  File "/home/alan/.ansible/tmp/ansible-1383244452.89-240799362355153/ec2", line 645, in main
    (instance_dict_array, new_instance_ids, changed) = create_instances(module, ec2)
  File "/home/alan/.ansible/tmp/ansible-1383244452.89-240799362355153/ec2", line 479, in create_instances
    ec2.create_tags(instids, module.from_json(instance_tags))
  File "/home/alan/.ansible/tmp/ansible-1383244452.89-240799362355153/ec2", line 1402, in from_json
    return json.loads(data)
  File "/usr/lib/python2.7/json/__init__.py", line 328, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 365, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer


FATAL: all hosts have already failed -- aborting
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


           to retry, use: --limit @/home/alan/site.retry

127.0.0.1                  : ok=1    changed=0    unreachable=0    failed=1
```

This is the result when I run it with the previous commit https://github.com/ansible/ansible/commit/b5c9d5a59b32b25ff1d0bc9448a92b0bf0656848

``` python
 _____________
< PLAY [test] >
 -------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


 _________________
< GATHERING FACTS >
 -----------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


ok: [127.0.0.1]
 _______________________
< TASK: launch instance >
 -----------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


changed: [127.0.0.1]
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


127.0.0.1                  : ok=2    changed=1    unreachable=0    failed=0 
```

**Comment by mpdehaan** [comment]
What I really want here is to not require JSON inside and just be able to take basic tag structures.

The idea that we have to embed JSON in Ansible content to use the tags option is offensive to me and our simplicity goals.

If we can solve this by instead supporting a non-JSON way this should be fine.

**Comment by mpdehaan** [comment]
@jctanner and @jlaska -- can you debug this one?  If you get a dictionary type here you would not have to do json.loads and this might be very simple.

I see we have in the argument spec this however:

```
        instance_tags = dict(),
```

What I really would like is a "type='dict'" parameter to instance_tags that works just like 'list' but can take JSON, a repr printed dict, or even key=value and works correctly regardless.

(NOTE:  I may look into this now)

**Comment by mpdehaan** [comment]
I see we have type=dict BTW it just needs to get smarter

**Comment by mpdehaan** [comment]
Possible fix pushed in 4318b27 and 08f42d3

Needs further testing but let us know if this works for you.

**Comment by dekked** [comment]
Issue is definitely fixed now :+1: 

Thank you!
